### PR TITLE
Use node-config for configuration

### PR DESCRIPTION
**Pull request by davidmz**
This PR replaces our old configuration management to `node-config`. It has several important changes:
1. The `./config` folder is cleaned up and there are only config files left in it. Other files was moved to `./app/setup`.
2. App config is a single source of truth now. The `knexfile.js` is not gitignored anymore and exposes data from the app config.
3. There is a separate config for Travis CI environment: `./config/default-travis.json`.

### CODE REVIEW

File: app/setup/environment.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ exports.init = async function (app) {
     await next();
   });
 
-  const accessLogStream = fs.createWriteStream(`${__dirname}/../log/${env}.log`, { flags: 'a' });
+  const accessLogStream = fs.createWriteStream(`${__dirname}/../../log/${env}.log`, { flags: 'a' });
```
Comment (id) [inline app/setup/environment.js:105]: would be great to have log directory also configurable

Comment (davidmz) [reply app/setup/environment.js:105]: Should we support relative paths here? Now the default is "./logs" relative to the root of the repository (not even cwd).

**Comment by id** [comment]
Complementary PR: https://github.com/FreeFeed/freefeed-ansible/pull/55

File: app/setup/environment.js (hunk 1 above)
Comment (id) [reply app/setup/environment.js:105]: @davidmz what I would normally do: get process working directory and use `${workdir}/logs` as the default for `logdir`.
And allow to override `logdir` via config/env.

**Pull request merged**